MobileBuy/buy/src/main/java/com/shopify/buy/dataprovider/AddressServiceDefault.java

```diff
@@ -77,6 +81,33 @@ public CancellableTask createAddress(final Long customerId, final Address addres
             .observeOn(callbackScheduler);
     }
 
+    public CancellableTask deleteAddress(Long customerId, Long addressId, Callback<Void> callback) {
+        return new CancellableTaskSubscriptionWrapper(deleteAddress(customerId, addressId).subscribe(new InternalCallbackSubscriber<>(callback)));
+    }
+
+    public Observable<Void> deleteAddress(Long customerId, Long addressId) {
+        if (customerId == null) {
+            throw new NullPointerException("customerId cannot be null");
+        }
+
+        if (addressId == null) {
+            throw new NullPointerException("addressId cannot be null");
+        }
+
+        int[] successCodes = {HTTP_NO_CONTENT};
+
+        return retrofitService.deleteAddress(customerId, addressId)
+            .doOnNext(new RetrofitSuccessHttpStatusCodeHandler<>(successCodes))
+            .map(new Func1<Response<Void>, Void>() {
```

Comment:
lol, oh ya. will do


On Tue, May 31, 2016 at 11:37 AM Ivan Savytskyi <notifications@github.com>
wrote:

> In
> MobileBuy/buy/src/main/java/com/shopify/buy/dataprovider/AddressServiceDefault.java
> <https://github.com/Shopify/mobile-buy-sdk-android/pull/162#discussion_r65207101>
> :
>
> > +    }
> > +
> > +    public Observable<Void> deleteAddress(Long customerId, Long addressId) {
> > +        if (customerId == null) {
> > +            throw new NullPointerException("customerId cannot be null");
> > +        }
> > +
> > +        if (addressId == null) {
> > +            throw new NullPointerException("addressId cannot be null");
> > +        }
> > +
> > +        int[] successCodes = {HTTP_NO_CONTENT};
> > +
> > +        return retrofitService.deleteAddress(customerId, addressId)
> > +            .doOnNext(new RetrofitSuccessHttpStatusCodeHandler<>(successCodes))
> > +            .map(new Func1<Response<Void>, Void>() {
>
> return response.body(); :)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Shopify/mobile-buy-sdk-android/pull/162/files/2aae9de514bf8b98f190a47786fd0a877931c825#r65207101>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABqEwP9F5c6CICr_QoxgBQZ8dzXDNy9Sks5qHFWlgaJpZM4IqlGp>
> .
>